ddtrace/contrib/internal/openai_agents/processor.py

```diff
@@ -0,0 +1,124 @@
+from typing import Any
+
+import agents
+from agents.tracing.processor_interface import TracingProcessor
+from agents.tracing.spans import Span as OaiSpan
+from agents.tracing.traces import Trace as OaiTrace
+
+from ddtrace.internal.logger import get_logger
+from ddtrace.llmobs._integrations.utils import OaiSpanAdapter
+from ddtrace.llmobs._integrations.utils import OaiTraceAdapter
+from ddtrace.trace import Pin
+
+
+logger = get_logger(__name__)
+
+
+class LLMObsTraceProcessor(TracingProcessor):
+    def __init__(self, integration):
+        super().__init__()
+        self.integration = integration
+
+    def on_span_start(self, span: OaiSpan[Any]) -> None:
+        """Called when a span starts.
+
+        Args:
+            span: The span that started.
+        """
+        self.integration.start_span_from_oai_span(Pin.get_from(agents), oai_span=OaiSpanAdapter(span))
+
+    def on_trace_start(self, trace: OaiTrace) -> None:
+        """Called when a trace starts.
+
+        Args:
+            trace: The trace that started.
+        """
+        self.integration.start_span_from_oai_trace(Pin.get_from(agents), oai_trace=OaiTraceAdapter(trace))
+
+    def on_trace_end(self, trace: OaiTrace) -> None:
+        """Called when a trace is finished.
+
+        Args:
+            trace: The trace that started.
+        """
+        trace_adapter = OaiTraceAdapter(trace)
+        cur_trace = self.integration.oai_to_llmobs_span.get(trace_adapter.trace_id)
```

Comment:
```suggestion
        trace_root_span = self.integration.oai_to_llmobs_span.get(trace_adapter.trace_id)
```
I think the naming is a bit confusing here, we are creating/referencing a span to represent the actual trace itself, not referencing the literal trace right?

Reply:
yea it's a reference to the trace root span so your suggestion makes the naming clearer - will apply